src/System.Text.Json/tests/NewtonsoftTests/CustomObjectConverterTests.cs

```diff
@@ -0,0 +1,213 @@
+            Assert.Equal(0, deserialized.DecimalRange.First);
+            Assert.Equal(1, deserialized.DecimalRange.Last);
+            Assert.Equal(int.MinValue, deserialized.IntRange.First);
+            Assert.Equal(int.MaxValue, deserialized.IntRange.Last);
+            Assert.Equal(null, deserialized.NullDecimalRange);
+            Assert.Equal(2010, deserialized.Year);
+        }
+
+        [Fact]
+        public void DeserializeByteArrayFromJsonArray()
+        {
+            string json = @"{
+  ""ByteArray"": ""AAECAw=="",
+  ""NullByteArray"": null
+}";
+
+            ByteArrayClass c = JsonSerializer.Parse<ByteArrayClass>(json);
+            Assert.NotNull(c.ByteArray);
+            Assert.Equal(4, c.ByteArray.Length);
+            Assert.Equal(new byte[] { 0, 1, 2, 3 }, c.ByteArray);
+        }
+
+        [Fact]
+        public void SerializeByteArrayClass()
+        {
+            ByteArrayClass byteArrayClass = new ByteArrayClass();
+            byteArrayClass.ByteArray = s_testData;
+            byteArrayClass.NullByteArray = null;
+
+            string json = JsonSerializer.ToString(byteArrayClass, new JsonSerializerOptions { WriteIndented = true });
+
+            Assert.Equal(@"{
+  ""ByteArray"": ""VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ=="",
+  ""NullByteArray"": null
+}", json);
+        }
+
+        [Fact]
+        public void DeserializeByteArrayClass()
+        {
+            string json = @"{
+  ""ByteArray"": ""VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ=="",
+  ""NullByteArray"": null
+}";
+
+            ByteArrayClass byteArrayClass = JsonSerializer.Parse<ByteArrayClass>(json);
+
+            Assert.Equal(s_testData, byteArrayClass.ByteArray);
+            Assert.Equal(null, byteArrayClass.NullByteArray);
+        }
+
+        private static readonly byte[] s_testData = Encoding.UTF8.GetBytes("This is some test data!!!");
+
+        [Fact]
+        public void AssertShouldSerializeTest()
+        {
+            MyClass myClass = new MyClass
+            {
+                Value = "Foo",
+                Thing = new MyThing { Number = 456, }
+            };
+            string json = JsonSerializer.ToString(myClass);
+
+            const string expected = @"{""Value"":""Foo"",""Thing"":{""Number"":456}}";
+            Assert.Equal(expected, json);
+        }
+
+        [Fact]
+        public void AssertDoesNotDeserializeInterface()
+        {
+            const string json = @"{
+""Value"": ""A value"",
+""Thing"": {
+""Number"": 123
+}
+}";
+            NotSupportedException e = Assert.Throws<NotSupportedException>(() =>
+            {
+                JsonSerializer.Parse<List<MyClass>>(json);
+            });
+            Assert.Equal("Deserialization of interface types is not supported. Type 'System.Text.Json.Tests.IThing'", e.Message);
+        }
+    }
+
+    internal class Range<T>
+    {
+        public T First { get; set; }
+        public T Last { get; set; }
+    }
+
+    internal class NullInterfaceTestClass
```

Comment:
Why are some of these internal while others public?

Reply:
imported as is ...